pysat/instruments/methods/general.py

```diff
@@ -108,7 +106,7 @@ def convert_timestamp_to_datetime(inst, sec_mult=1.0, epoch_name='Epoch'):
     """
 
     inst.data[epoch_name] = pds.to_datetime(
-        [dt.datetime.utcfromtimestamp(x * sec_mult)
+        [dt.datetime.utcfromtimestamp(int(x * sec_mult))
```

Comment:
```suggestion
        [dt.datetime.utcfromtimestamp(int(np.floor(x * sec_mult)))
```

Ok, what you have is fine.  However, since python 3 decided to change to way that integer division works, I have been being extra specific in my casting after doing arithmetic.  So, this is a suggestion that there are good reasons to either take or leave.  I am ok either way, but want to consider the behaviour.

Reply:
Hmm.  trying both the suggestion and `np.floor().astype(int)` don't seem to fix the problem.  Reading up on the issue, it seems that there are potential bugs in using the naive construction (`utcfromtimestamp`) rather than an aware construction with a specified time zone.  II'm going to leave this as is for now, but write up an issue.

Reply:
Oops.  Copied this over wrong in my local time.  Trying again.

Reply:
This works.  Updating.